qiskit/synthesis/__init__.py

```diff
@@ -180,8 +180,12 @@
     two_qubit_cnot_decompose,
     TwoQubitWeylDecomposition,
 )
-from .multi_controlled.mcx_with_ancillas_synth import (
+from .multi_controlled import (
     synth_mcx_n_dirty_i15,
     synth_mcx_n_clean_m15,
     synth_mcx_1_clean_b95,
+    synth_mcx_mcphase,
+    synth_mcx_gray_code,
+    synth_c3x,
+    synth_c4x,
```

Comment:
I don't think we would like to add synt_c3x and synt_c4x to the docs (since we don't want to copy the circuit library into the synthesis library).
in fact, why can't we use the C3XGate and C4XGate directly?

Reply:
I believe that the actual synthesis methods should be in the synthesis library and not in the circuit library (as part of gates' `definition`s).

Some other things to consider:
* Do we want to reuse these functions from other places? I believe we might.
* What if at some point we will find better decompositions for 3- and 4- controlled X-gates? I think changing the synthesis methods is easier than going inside the gate's definition.
* Do we want to remove C3X and C4X gates altogether? I don't think we have plans for that, but this point periodically comes up in offline discussions.